### PR TITLE
[stdlib] Documentation revisions

### PR DESCRIPTION
**Pull request by natecook1000**
- Update NSRange -> Range guidance
- Fix example in Optional
- Improve RangeExpression docs
- Fix issue in UnsafeRawBufferPointer.initializeMemory
- Code point -> scalar value most places
- Reposition the dot above the scripty `i'
- Fix ExpressibleByArrayLiteral code sample

### CODE REVIEW

**Comment by natecook1000** [comment]
@swift-ci Please smoke test

File: stdlib/public/core/Range.swift.gyb [hunk 1]
```diff
@@ -10,11 +10,14 @@
...
 
-/// A type which can be used to slice a collection. A `RangeExpression` can
-/// convert itself to a `Range<Bound>` of indices within a given collection;
-/// the collection can then slice itself with that `Range`.
+/// A type that can be used to slice a collection.
+///
+/// A type that conforms to `RangeExpression` can convert itself to a
+/// `Range<Bound>` of indices within a given collection.
 public protocol RangeExpression {
+  /// The type that an expression describes a range of.
```
Comment (xwu) [inline stdlib/public/core/Range.swift.gyb]: Suggestion: how about the following?

> The type of elements for which the expression describes a range.

File: stdlib/public/core/UnicodeScalar.swift [hunk 2]
```diff
@@ -109,39 +109,39 @@ extension Unicode.Scalar :
...
   ///     if let bap = Unicode.Scalar(codepoint) {
-  ///       print(bap)
+  ///         print(bap)
   ///     } else {
-  ///       // Do something else
+  ///         // Do something else
   ///     }
+  ///
+  /// - Parameter v: The Unicode code point to use for the scalar. The
+  ///   initializer succeeds if `v` is a valid Unicode scalar value, in the
```
Comment (xwu) [inline stdlib/public/core/UnicodeScalar.swift]: Nit: no comma.

**Comment by natecook1000** [comment]
@swift-ci Please smoke test and merge

**Pull request merged**